entities/cb_game.rb

```diff
@@ -0,0 +1,152 @@
+class CBGame
+  include CodebreakerCaptainjns
+  include DataUtils
+
+  TOTAL_SIGNS = 4
+
+  def self.call(env)
+    new(env).response.finish
+  end
+
+  def initialize(env)
+    @request = Rack::Request.new(env)
+  end
+
+  def response
+    case @request.path
+    when '/' then main
+    when '/rules' then Rack::Response.new(render('rules.html.erb'))
+    when '/statistics' then Rack::Response.new(render('statistics.html.erb'))
+    when '/start' then start
+    when '/play_again' then play_again
+    when '/check' then check
+    when '/lose' then lose
+    when '/game' then game
+    when '/win' then win
+    when '/show_hint' then show_hint
+    else Rack::Response.new('Not Found', 404)
+    end
+  end
+
+  private
+
+  def main
+    return redirect('game') if @request.session.key?(:game)
+
+    Rack::Response.new(render('menu.html.erb'))
+  end
+
+  def play_again
+    @request.session.delete(:game)
+
+    redirect
+  end
+
+  def start
+    start_redirect || start_initialize
+  end
+
+  def game
+    game_redirect || Rack::Response.new(render('game.html.erb'))
+  end
+
+  def show_hint
+    return redirect unless @request.session.key?(:game)
+
+    @request.session[:hints] << @request.session[:game].use_hint if @request.session[:game].hints.positive?
+    redirect('game')
+  end
+
+  def check
+    return redirect unless @request.params.key?('number')
+
+    @request.session[:result] = @request.session[:game].check(@request.params['number'])
+
+    if @request.session[:game].win
+      save_results unless @request.session[:save]
+      return redirect('win')
+    end
+
+    redirect('game')
+  end
+
+  def lose
+    lose_redirect || Rack::Response.new(render('lose.html.erb'))
+  end
+
+  def win
+    win_redirect || Rack::Response.new(render('win.html.erb'))
+  end
+
+  def render(template)
+    path = File.expand_path("../../templates/#{template}", __FILE__)
+    ERB.new(File.read(path)).result(binding)
+  end
+
+  def redirect(address = '')
+    Rack::Response.new { |response| response.redirect("/#{address}") }
+  end
+
+  def save_results
+    save(summary)
```

Comment:
Why do you need this method? Seems that you can use just `save(summary)` instead of this wrapper.